DIPs/DIP.md

```diff
@@ -0,0 +1,123 @@
+# In-place struct initialization
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1002                                                            |
+| Author:         | Cédric Picard (cpicard@openmailbox.org)                         |
+| Implementation: | N/A                                                             |
+| Status:         | N/A                                                             |
+
+## Abstract
+
+Structs support static initialization. This DIP proposes to extend this
+initialization to in-place struct declaration. This change could notably be
+used to mimic keyword arguments and make argument manipulations at runtime
+easier.
+
+### Links
+
+- [Language specs on structs](https://dlang.org/spec/struct.html)
+
+## Description
+
+Let S be a struct defined as:
+
+    struct S {
+        uint a;
+        long b;
+        int  c;
+    }
+
+The proposed change is to make the following syntax legal
+
+    auto s = S(a:42, b:-5);
+
+and in all points equivalent to what follows
+
+    S s = { a:42, b:-5 };
+
+This equivalence standing, the following must compile:
+
+    S s = S(b:-5);
+    S s = S(b: -5, a :42);
```

Comment:
From what I remember of skipping through the keyword threads, is that Walter didn't like arbitrary order.  It might increase your chances of approval, if you don't allow arbitrary order or at least provide good reasoning why it should be allowed ;-)